### PR TITLE
Build using JDK 11

### PR DESCRIPTION
**Pull request by pettyjamesm**
Targets JDK11 when building the artifact, allowing use of a JDK9+ API method for parsing `long` values without an extra `String` allocation and generates optimized string concatenations when serializing by avoiding an intermediate `StringBuilder` step (JDK9+ improvement in javac emitted code).

### CODE REVIEW

**Comment by pettyjamesm** [comment]
Follow up from https://github.com/airlift/units/pull/28 - change the build to target support for JDK 11 and no longer supporting JDK8 runtimes.

cc: @martint @electrum @findepi

File: src/main/java/io/airlift/units/DataSize.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public DataSize convertToMostSuccinctDataSize()
     @JsonValue
     public String toBytesValueString()
     {
-        return Long.toString(bytes) + Unit.BYTE.getUnitString();
+        return bytes + Unit.BYTE.getUnitString();
```
Comment (findepi) [inline src/main/java/io/airlift/units/DataSize.java:212]: Looks unrelated to "Avoid extra string allocation in DataSize parsing". Separate commit?

**Comment by findepi** [comment]
> Follow up from #28

cc @nineinchnick

File: src/main/java/io/airlift/units/DataSize.java (hunk 1 above)
Comment (pettyjamesm) [reply src/main/java/io/airlift/units/DataSize.java:212]: Updated commit message to mention this optimization as well instead.

**Comment by pettyjamesm** [comment]
@nineinchnick should we consider your 👍 on the PR description to be an approval or do you need more time to take a look at the change?

cc @findepi

**Comment by nineinchnick** [comment]
> @nineinchnick should we consider your 👍 on the PR description to be an approval or do you need more time to take a look at the change?

I don't know enough about this library to approve these changes. I just expressed my general support for the idea. I've been mentioned, so I felt obliged.

**Comment by pettyjamesm** [comment]
@nineinchnick, sure thing- thanks for letting me know!

@findepi - are we good with the approach of requiring JDK11+ at runtime for all future releases of the units library? If so, anything else we need to do before merging?

**Comment by electrum** [comment]
Thanks!

**Comment by pettyjamesm** [comment]
@electrum - thanks for the merge! We should probably start a new version (major version?) since we know about the compatibility break in the next release at this point

**Comment by pettyjamesm** [comment]
> `trino-jdbc` uses units
> 
> ```
> trino master$ mvnd dependency:tree -pl :trino-jdbc | grep units
> [INFO] +- io.airlift:units:jar:1.7:compile
> ```
> 
> and trino-jdbc still builds for Java 8 https://github.com/trinodb/trino/blob/779e0f5ac98b00b50eb9077edee846f46e1d32bb/client/trino-jdbc/pom.xml#L17

That’s correct, which is why I both asked whether this change should come with incrementing the major version number (it’s not totally backwards compatible, after all) and asked whether the trino clients should start requiring JDK11+ moving forward on the Trino PR that split the units dependency version between clients and servers but didn’t get a response to either.

In any case, the state today is indeed that Trino clients are using version 1.7 to maintain JDK8 support and the server is using 1.8 since it already requires JDK17.
